contrib/mod_sftp/fxp.c

```diff
@@ -4314,7 +4314,7 @@ static int fxp_handle_ext_check_file(struct fxp_packet *fxp, char *digest_list,
   }
 
   cmd = fxp_cmd_alloc(fxp->pool, "SITE_DIGEST", pstrdup(fxp->pool, path));
-  if (!dir_check(fxp->pool, cmd, "READ", path, NULL)) {
+  if (!dir_check(fxp->pool, cmd, "READFILE", path, NULL)) {
```

Comment:
This means that `<Limit READ>` configurations, for SFTP sessions, would no longer match/work as expected.

Reply:
Hmm...  OK I can see that.  But this feels like it's one of those things that should have been done differently from the beginning since there was already a class named READ and WRITE, and there may not be a way to resolve the issue without a breaking change.  The one thing I can think of would be to either have a compile-time directive of whether to use READ or READFILE (and WRITE or WRITEFILE), or maybe even a runtime directive, which would allow configurable backward compatibility until people were ready to make the switch.  And it would give you time to officially deprecate the READ/WRITE command names and ultimately remove them in a future version.  Because as it stands right now, you cannot control the READ/WRITE commands individually in anything that also accepts command classes, because using them there affects the entire class rather than just the command...

Actually, I just realized that the documentation on Limit explicitly refers to the classes and not the commands:
```
    What about the READ and WRITE command groups?

    <Limit READ>
    Applies to the SYMLINK and READ SFTP requests

    <Limit WRITE>
    Applies to the LOCK, SETSTAT, SYMLINK, and WRITE SFTP requests
```
So based on the current documentation, this change would not actually break anything, because the SFTP READ(FILE) command would still be affected by a limit on the READ class, and the same for WRITE(FILE) and the WRITE class.

Your argument below about logfile parsers is definitely more impactful, though based on what the READ and WRITE commands currently log I'm not sure how useful they are for parsing vs the rest of the logs, and depending on what they are looking for if all they look for is READ and WRITE they might still match since only SFTP uses those commands in the logs.  I don't know what the right way to make a transition would be, but it feels pretty solid that the commands READ and WRITE need to somehow be differentiated from the classes READ and WRITE, and since the classes came first (and encompass the relevant commands), its the commands that should be renamed and not the classes.